File: src/models/payment/seller-payment-method.ts
interface ISellerPaymentMethod {
    seller_payment_method_id: number;
    payment_method_id: number;
    is_default: boolean;
    text: string;
    expire_date?: string;
    cardholder_name: string;
}

class SellerPaymentMethod implements ISellerPaymentMethod {
    constructor(
        public seller_payment_method_id: number,
        public payment_method_id: number,
        public is_default: boolean,
        public text: string,
        public cardholder_name: string,
        public expire_date?: string // Optional in the second constructor
    ) {}

    static fromJSON(data: any): SellerPaymentMethod {
        return new SellerPaymentMethod(
            data.seller_payment_method_id,
            data.payment_method_id,
            data.is_default,
            data.text,
            data.cardholder_name,
            data.expire_date // Optional expire_date
        );
    }
}

File: backend/API/Seller/Payment/Data/Models/SellerPaymentMethod.cs
namespace POD.API.Seller.Payment.Data.Models
{
	public class SellerPaymentMethod
	{
        public SellerPaymentMethod( int sellerPaymentMethodId, int paymentMethodId, bool isDefault, string text, 
            string expireDate, string cardholderName)
        {
            SellerPaymentMethodId = sellerPaymentMethodId;
            PaymentMethodId = paymentMethodId;
            IsDefault = isDefault;
            Text = text;
            ExpireDate = expireDate;
            CardholderName = cardholderName;
        }
        
        public SellerPaymentMethod( int sellerPaymentMethodId, int paymentMethodId, bool isDefault, string text,
            string cardholderName)
        {
            SellerPaymentMethodId = sellerPaymentMethodId;
            PaymentMethodId = paymentMethodId;
            IsDefault = isDefault;
            Text = text;
            CardholderName = cardholderName;
        }

        public int SellerPaymentMethodId { get; set; }
        public int PaymentMethodId { get; set; }
        public bool IsDefault { get; set; }
        public string Text { get; set; }
        public string ExpireDate { get; set; }
        public string CardholderName { get; set; }
    }
}

